component/src/main/java/org/wso2/extension/siddhi/store/ignite/ApacheIgniteStore.java

```diff
@@ -0,0 +1,1158 @@
+                                "exists already with 'symbol' as the primary key.The connection is made as specified" +
+                                "by the parameters configured under '@Store' annotation.Data is inserted into table," +
+                                "stockTable from stockStream"
+                ),
+                @Example(
+                        syntax = "define stream StockStream (symbol string, price float, volume long);\n " +
+                                "@Store(type=\"apacheignite\", url = \" jdbc:ignite:thin://127.0.0.1 \" ," +
+                                "username=\"ignite \", password=\" ignite ) \n" +
+                                "@PrimaryKey(\"symbol\")\n" +
+                                "define table StockTable (symbol string, price float, volume long);\n" +
+                                "@info(name = 'query2')\n " +
+                                "from FooStream#window.length(1) join StockTable on " +
+                                "StockTable.symbol==FooStream.name \n" +
+                                "select StockTable.symbol as checkName, " +
+                                "StockTable.volume as checkVolume," +
+                                "StockTable.price as checkCategory\n " +
+                                "insert into OutputStream;",
+                        description = "The above example creates a table in apache ignite data store if it does not " +
+                                "exists already with 'symbol' as the primary key.The connection is made as specified" +
+                                "by the parameters configured under '@Store' annotation.Then the table is joined with" +
+                                " a stream name 'FooStream' based on a condition. The following operations are " +
+                                "included in the condition:\n" +
+                                "[AND, OR, Comparisons(<, <=, >, >=, ==, != )]"
+                )
+        }
+)
+
+public class ApacheIgniteStore extends AbstractQueryableRecordTable {
+
+    private static final Log log = LogFactory.getLog(ApacheIgniteStore.class);
+    private String tableName;
+    private String url;
+    private String username;
+    private String password;
+    private String schema;
+    private String template;
+    private String distributeJoins;
+    private String enforceJoinOrder;
+    private String collocated;
+    private String replicatedOnly;
+    private String autocloseServerCursor;
+    private String socketSendBuffer;
+    private String socketReceiveBuffer;
+    private String backups;
+    private String atomicity;
+    private String affinityKey;
+    private String cacheName;
+    private String dataRegion;
+    private StringBuilder connectionParams;
+    private boolean connected;
+    private boolean isAuthEnabled;
+    private Annotation storeAnnotation;
```

Comment:
Why are the annotation stored as class variable? Ideall, this should not be needed after initialisation, refactor code